src/init.cpp

```diff
@@ -1887,7 +1887,7 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
             if (!zaddr.has_value()) {
                 return InitError(_("-mineraddress is not a valid " PACKAGE_NAME " address."));
             }
-            auto ztxoSelector = pwalletMain->ZTXOSelectorForAddress(zaddr.value(), true, false);
+            auto ztxoSelector = pwalletMain->ZTXOSelectorForAddress(zaddr.value(), true, false, TransactionStrategy(PrivacyPolicy::FullPrivacy));
```

Comment:
This is what I meant in my earlier comment: `FullPrivacy` is misleading here because coinbase transactions can never satisfy that policy, but the policy here is actually irrelevant as long as it is not `AllowLinkingAccountAddresses` or `NoPrivacy`. And in the context of the `-mineraddress` flag, all we want to do is confirm that we have a spending key corresponding to the address, so address linkability is irrelevant (because by the time we check it, we've already confirmed we have an account corresponding to the address).

Reply:
Ah, yeah. There should perhaps be a constant or something like `minerTxStrategy` or `shieldCoinbaseStrategy` with `AllowRevealedSenders` as the policy, and we could use that in contexts like this one.

Reply:
See https://github.com/zcash/zcash/pull/6377#discussion_r1128838809